mongodb-setup/app/server/db.server.ts

```diff
@@ -0,0 +1,37 @@
+import * as dotenv from 'dotenv';
+import mongoose, { Schema } from 'mongoose';
+import { UserModel } from '~/types';
+
+dotenv.config();
+
+let db: any;
+
+connect();
+
+async function connect() {
+    try {
+        if (!process.env.MONGODB_URL) throw new Error('MONGODB_URL is required');
+
+        if (process.env.NODE_ENV === 'production') {
+            db = await mongoose.connect(process.env.MONGODB_URL);
+        } else {
+            if (!(global as any).__db) {
+                (global as any).__db = await mongoose.connect(process.env.MONGODB_URL);
+            }
+            db = (global as any).__db;
+        }
+        return db;
+    } catch (error: any) {
+        console.error('--- ERROR', error);
+    }
+}
+
+// Users
+const UserSchema: Schema = new Schema<UserModel>({
```

Comment:
No need for explicit typing here

```suggestion
const UserSchema = new Schema<UserModel>({
```